src/test/java/seedu/address/model/good/GoodTest.java

```diff
@@ -63,6 +69,10 @@ public void equalsTest() {
         // different good quantity -> returns false
         editedApple = new GoodBuilder(APPLE).withGoodQuantity(VALID_GOOD_QUANTITY_ONE).build();
         assertFalse(APPLE.equals(editedApple));
+
+        // different threshold -> returns false
+        editedApple = new GoodBuilder(APPLE).withThreshold(VALID_GOOD_QUANTITY_ONE).build();
+        assertFalse(APPLE.equals(editedApple));
```

Comment:
Should this be the case? Suppose I buy a good, will it just create duplicate goods?

Reply:
When buying/adding a good, a less restricted method isSameGood is called. isSameGood only checks for the goodname, and it is a weaker notation of equality between 2 goods. While the equals method is stronger notation of equality and require more stringent checks.